module/lua/setjmp/setjmp_x86_64.S

```diff
@@ -44,11 +53,11 @@ x:
 
 #define	SET_SIZE(x) \
 	.size	x, [.-x]
-#endif	// WIN32
 
-#if defined(__linux__) && defined(CONFIG_SLS)
-#define	RET	ret; int3
-#else
+#endif
+
+
+#ifndef RET
 #define	RET	ret
 #endif
 
```

Comment:
Here is where it gets fun.
Unix: the first 6 parameters passed to a function are:
rdi, rsi, rdx, rcx, r8, r9, (7th and onwards are on stack)
Windows: the first 4 parameters passed to a function are:
rcx, rdx, r8, and r9 (5th and onwards are on stack)

Problem 1: argument use of registers need to be handled. setjmp(env) only takes one arg (rdi vs rcx), and longjmp(env, val) takes two. (rdi, rsi) -> (rcx, rdx).

Problem 2: rdi and rsi needs to be saved (non-volatile). So often Unix assembler will use rdi and rsi as scratch registers.

Problem 3: Similar thing with xmm registers, need to save xmm6 and xmm15.

This is why I've kept separate ASM for Win so far.  There probably is a nice way to handle the differences
but I don't work enough in assembler to be sure :)